tests/templates/test_integration.py

```diff
@@ -176,558 +138,347 @@
-                 'k': cvqnn_layers_kappa_normal(n_layers=3, n_wires=2),
-                 'wires': range(2)}),
-               (Interferometer,
-                {'phi': interferometer_phi_uniform(n_wires=2), 'varphi': interferometer_varphi_uniform(n_wires=2),
-                 'theta': interferometer_theta_uniform(n_wires=2), 'wires': range(2)}),
-               (Interferometer,
-                {'phi': interferometer_phi_normal(n_wires=2), 'varphi': interferometer_varphi_normal(n_wires=2),
-                 'theta': interferometer_theta_normal(n_wires=2), 'wires': range(2)}),
-               (Interferometer,
-                {'phi': interferometer_phi_uniform(n_wires=3), 'varphi': interferometer_varphi_uniform(n_wires=3),
-                 'theta': interferometer_theta_uniform(n_wires=3), 'wires': range(3)}),
-               (Interferometer,
-                {'phi': interferometer_phi_normal(n_wires=3), 'varphi': interferometer_varphi_normal(n_wires=3),
-                 'theta': interferometer_theta_normal(n_wires=3), 'wires': range(3)})
-               ]
-
-    @pytest.mark.parametrize("template, dict", QUBIT_INIT)
-    def test_integration_qubit_init(self, template, dict):
-        """Checks parameter initialization compatible with qubit templates."""
-
-        n_wires = len(dict['wires'])
-        dev = qml.device('default.qubit', wires=n_wires)
 
         @qml.qnode(dev)
-        def circuit():
-            template(**dict)
-            return qml.expval(qml.Identity(0))
+        def circuit(nondiffable=None):
+            # Add wires
+            nondiffable['wires'] = range(2)
+
+            # Circuit with operations before and
+            # after the template is called
+            if op_before_template:
+                QubitTemplate(w=0)
+                qml.PauliX(wires=0)
+            template(**nondiffable)
+            if not op_before_template:
+                QubitTemplate(w=0)
+                qml.PauliX(wires=1)
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
 
         # Check that execution does not throw error
-        circuit()
+        circuit(nondiffable=nondiffable)
 
-    @pytest.mark.parametrize("template, dict", CV_INIT)
-    def test_integration_qubit_init(self, template, dict, gaussian_dummy):
-        """Checks parameter initialization compatible with qubit templates."""
+    @pytest.mark.parametrize("op_before_template", [True, False])
+    @pytest.mark.parametrize("template, diffable, nondiffable", CV_DIFFABLE_NONDIFFABLE)
+    def test_integration_cv_nondiffable(self, template, diffable, nondiffable, gaussian_device_2_wires,
+                                        op_before_template):
+        """Tests integration of cv templates passing differentiable arguments as auxiliary arguments to qnode."""
 
-        n_wires = len(dict['wires'])
-        dev = gaussian_dummy(n_wires)
+        # Change type of differentiable arguments
```

Comment:
```suggestion
        # skip this test if template does not allow for operations before
        if template.__name__ in NO_OP_BEFORE and op_before_template:
            pytest.skip("Template does not allow operations before - skipping this test.")
            
        # Change type of differentiable arguments
```
Copying from the qubit case? I guess now there aren't any CV templates in `NO_OP_BEFORE`, but maybe there will be in future?

Reply:
I'd prefer if people who add tests adapt the structure to their needs...

Reply:
Ok, up to you, I think it's good to minimize the overhead - I guess it depends how likely we expect a CV template to come in soon that doesn't allow operations before it.